src/components/container.js

```diff
@@ -1,38 +1,96 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 
+
 const Container = class Container extends Component {
   static propTypes = {
     title: PropTypes.string.isRequired,
     children: PropTypes.array,
   };
+
   render() {
     const { title, children } = this.props;
-    const container = {
+
+    const popUpBox = {
+      width: "100%",
```

Comment:
Once the infinite recursion of Main and Routes is fixed, the modal windows do actually render, though they end up behind the map. To fix that, you'll need to set popUpBox's position to absolute (so z-index works on it) and its z-index to at least 3 so it's above the map and everything else.

After that, it's just a matter of getting the rest of popUpBox and container's styling right and giving it a way to be dismissed. If you set the popUpBox as width and height 100% and a translucent black background, you can use it as the clickable dim shield while container holds the actual modal content.